src/resolve/color.ts

```diff
@@ -125,5 +113,34 @@ function hexToRgba(hex: string): string {
   return `rgba(${r}, ${g}, ${b}, 1)`;
 }
 
+function findColorInConfigRecursive(
+  colorName: string,
+  config: TwColors
+): string {
+  const configColor = config[colorName];
+
+  // the color is found at the current config level
+  if (isString(configColor)) {
+    return configColor;
+  } else if (isObject(configColor) && isString(configColor.DEFAULT)) {
+    return configColor.DEFAULT;
+  }
+
+  // search for a matching sub-string at the current config level
+  let [colorNameStart = ``, ...colorNameRest] = colorName.split(`-`);
+  while (colorNameStart !== colorName) {
```

Comment:
let's say we have some long, invalid color utility like `bg-foo-bar-baz-not-real-nope-noway`.  if we end up with an `undefined` for `subConfig` we can break out of the while loop immediately, right? once we get an undefined, there's no reason to keep iterating.  you're guarding against not calling the method recursively, but can't we just immediately return if we get an `undefined` for `subObject`? we'll never then find a sub-subObject at that point, i think, right?

Reply:
The point of the while-loop is to support configs containing dashes `-` at any level of the config, potentionally reducing the need for nesting in the configuration.
Take the following config for example:
```javascript
{
    theme: {    
        colors: {
            'foo-bar': {
                DEFAULT: `#ff0000`,
                baz: `#EEF`
            },
        },
    },
}
```
If we where to break out of the while loop at the first sign of `undefined`, then the color `foo-bar-baz` would be unreachable. The algorithm would start with looking for `foo-bar-baz` at the root color level. As it's not found as a property of that object, it will continue to the while loop and try with the first part of the name: `foo`. 
Now `config['foo']` will be `undefined`. If we break out now, instead of keep trying with `foo-bar`, and `foo-bar-baz`(this last iteration can actually be skipped, as it has already been check at the start of the function), we never find the color. 
That beeing said, I'm not sure of the importance of supporting this. We could potentionally remove the entire while loop and just look for the sub-config of the first part of the name. Doing this, we still support configs where dashes appear at the leaf level, eg:
```javascript
{
    theme: {    
        colors: {
            foo: {
                bar: `#ff0000`,
                'bar-baz': `#EEF`
            },
        },
    },
}
```
The same results as the first config is achievable in this implementation as well, but it could require more nesting in the configuration. 
I'm quite unsure of whats best to be honest.

Reply:
Ok yup, that makes sense, thanks for the example.  

I fixed the CI stuff and added a commit also adding one more test, and a bailout point from the `while` loop (which probably is unnecessary, but while loops always make me nervous about getting stuck in an infinite loop).  I opened a PR to your PR. If you're good merging mine into yours, i'll merge yours into master and cut a release.

thanks again for the contribution. 👍

Reply:
Looked at your addition. The bail-out makes sense👍
Merged it now. 
Thanks for letting me contribute, it will make my work easier going forward 🙂